graph/pkg/service/v0/drives.go

```diff
@@ -553,3 +562,30 @@ func canSetSpaceQuota(ctx context.Context, user *userv1beta1.User) (bool, error)
 	}
 	return true, nil
 }
+
+func generateCs3Filters(request *godata.GoDataRequest) []*storageprovider.ListStorageSpacesRequest_Filter {
+	var filters []*storageprovider.ListStorageSpacesRequest_Filter
+	if request.Query.Filter != nil && request.Query.Filter.Tree.Token.Value == "eq" {
+		switch request.Query.Filter.Tree.Children[0].Token.Value {
+		case "driveType":
+			filter1 := &storageprovider.ListStorageSpacesRequest_Filter{
+				Type: storageprovider.ListStorageSpacesRequest_Filter_TYPE_SPACE_TYPE,
+				Term: &storageprovider.ListStorageSpacesRequest_Filter_SpaceType{
+					SpaceType: strings.Trim(request.Query.Filter.Tree.Children[1].Token.Value, "'"),
+				},
+			}
+			filters = append(filters, filter1)
+		case "id":
+			filter2 := &storageprovider.ListStorageSpacesRequest_Filter{
+				Type: storageprovider.ListStorageSpacesRequest_Filter_TYPE_ID,
+				Term: &storageprovider.ListStorageSpacesRequest_Filter_Id{
+					Id: &storageprovider.StorageSpaceId{
+						OpaqueId: strings.Trim(request.Query.Filter.Tree.Children[1].Token.Value, "'"),
+					},
+				},
+			}
+			filters = append(filters, filter2)
+		}
```

Comment:
Later, people will send queries with a filter, but an invalid operator. I am already dreaming of `filter lt 'Мне всё равно́.`. 

It would be great if the log contained a message such as `filter only supports the 'eq' operator so far.` to make it easier to find the problem.

Reply:
good idea.